### PR TITLE
Use Docc for documentation

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation

Documentation is nice, and we can help support users by providing useful
clear docs.

Modifications

Add Docc to 5.6 and later builds
Make sure symbol references work
Add overview docs

Result

Nice rendering docs

### CODE REVIEW

**Review by FranzBusch — APPROVED** [review]
LGTM!

File: Sources/StructuredFieldValues/Docs.docc/index.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# ``StructuredFieldValues``
```
Comment (glbrntt) [inline Sources/StructuredFieldValues/Docs.docc/index.md:1]: It seems like the first part of this which is duplicated in  `RawStructuredFieldVales` should be in a package overview document. Is that possible with docc?

**Review by glbrntt — APPROVED** [review]

File: Sources/RawStructuredFieldValues/Docs.docc/index.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+## About Structured Header Field Values
+
+HTTP Structured Header Field Values are a HTTP extension recorded in [RFC 8941](https://www.rfc-editor.org/rfc/rfc8941.html). They provide a set of data types and algorithms for handling HTTP header field values in a consistent way, allowing a single parser and serializer to handle a wide range of header field values.
+
+## Swift HTTP Structured Header Field Values
+
+This package provides a parser and serializer that implement RFC 8941. They are entirely complete, able to handle all valid HTTP structured header field values.
+
+This package provides two top-level modules: `StructuredFieldValues` and `RawStructuredFieldValues`.
```
[the remark this answers is not in the document]
Comment (ktoso) [reply Sources/RawStructuredFieldValues/Docs.docc/index.md:15]: nope :( File issues against them tho!

**Review by ktoso — APPROVED** [review]
LGTM overall 👍

File: Sources/StructuredFieldValues/Docs.docc/index.md (hunk 1 above)
Comment (Lukasa) [reply Sources/StructuredFieldValues/Docs.docc/index.md:1]: It is not, sadly, neither can I link the two together.

File: Sources/RawStructuredFieldValues/Docs.docc/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lukasa) [reply Sources/RawStructuredFieldValues/Docs.docc/index.md:15]: Nope.

**Pull request merged**